protocol/x/revshare/keeper/msg_update_unconditional_revshare_config.go

```diff
@@ -3,13 +3,31 @@ package keeper
 import (
 	"context"
 
+	errorsmod "cosmossdk.io/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/dydxprotocol/v4-chain/protocol/x/revshare/types"
 )
 
-// TODO(OTE-775): Add methods to set and get rev share config
 func (k msgServer) UpdateUnconditionalRevShareConfig(
 	goCtx context.Context,
 	msg *types.MsgUpdateUnconditionalRevShareConfig,
 ) (*types.MsgUpdateUnconditionalRevShareConfigResponse, error) {
-	return nil, nil
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if sender is authorized to set revenue share
+	if !k.HasAuthority(msg.Authority) {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority %s",
+			msg.Authority,
+		)
+	}
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+	k.SetUnconditionalRevShareConfigParams(ctx, msg.Config)
```

Comment:
Let's also add some stateful check per discussion [here](https://www.notion.so/RevShare-Generalization-6d89820112c94dfeaefc5a5ace052cf4?d=90f5509f43204245b948e5a8ca95f1ab&pvs=4#27fe68ecc6fd48539ca33eb818c34321). 

Let's create a new function 
`ValidateRevShareSafety`, which takes in:
- UnconditionalRevShareConfigParams
- [MarketMapperRevenueShareParams](https://github.com/dydxprotocol/v4-chain/blob/0ce92b3cd6448bd4eb06dde88f0bf098897f34fe/protocol/x/revshare/keeper/revshare.go#L11)
- AffiliateTiers

And check that: unconditional rev share % + market mapper rev share % + highest affiliate rev share % <= 100%

This can be done in a follow-up PR (pls add TODO item)